lib/matplotlib/widgets.py

```diff
@@ -401,9 +408,18 @@ def observers(self):
 
     def _value_in_bounds(self, val):
         """Makes sure *val* is with given bounds."""
-        if self.valstep:
+        if isinstance(self.valstep, Number):
             val = (self.valmin
                    + round((val - self.valmin) / self.valstep) * self.valstep)
+        elif self.valstep is not None:
+            valsnap = np.asanyarray(self.valstep)
```

Comment:
```suggestion
            valstep = np.asanyarray(self.valstep)
```
Introducing a new name signals that this is something else, but semantically this stays the same concept.